src/IssueViz/Editor/ErrorTagging/ErrorTagTooltipProvider.cs

```diff
@@ -0,0 +1,97 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.ComponentModel.Composition;
+using System.Windows.Controls;
+using System.Windows.Documents;
+using System.Windows.Media;
+using Microsoft.VisualStudio.PlatformUI;
+using Microsoft.VisualStudio.Shell;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Core.Analysis;
+using SonarLint.VisualStudio.Infrastructure.VS;
+
+namespace SonarLint.VisualStudio.IssueVisualization.Editor.ErrorTagging
+{
+    /// <summary>
+    /// Creates a tooltip object for an IErrorTag
+    /// </summary>
+    internal interface IErrorTagTooltipProvider
+    {
+        object Create(IAnalysisIssueBase analysisIssueBase);
+    }
+
+    [Export(typeof(IErrorTagTooltipProvider))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    internal class ErrorTagTooltipProvider : IErrorTagTooltipProvider
```

Comment:
If (and when) we drop VS 2015, we can do what roslyn do:
https://github.com/dotnet/roslyn/blob/afd10305a37c0ffb2cfb2c2d8446154c68cfa87a/src/EditorFeatures/Core/Implementation/Diagnostics/AbstractDiagnosticsAdornmentTaggerProvider.cs#L69

Unfortunately those classes are not available in 2015 and I couldn't find anything else.